src/pages/Lending/hooks/useRepaymentLockData.tsx

```diff
@@ -0,0 +1,92 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { getThorchainLendingPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainLending/utils'
+
+type UseLendingPositionDataProps = {
+  accountId?: AccountId
+  assetId?: AssetId
+}
+
+// non-exhaustive, we only use this to get the current blockheight
+type ThorchainBlock = {
+  header: {
+    height: number
+  }
+}
+
+export const useRepaymentLockData = ({ accountId, assetId }: UseLendingPositionDataProps) => {
+  const repaymentLockQueryKey: [
```

Comment:
seems like we can use the inferred type for the `queryKey` to clean this up a bit visually

Reply:
LFG! Technically fine in this case

<img width="849" alt="image" src="https://github.com/shapeshift/web/assets/17035424/7ca0d502-354a-401b-8956-51fe014c0614">

Was going with this pattern as I was previously losing type narrowing in some other react-query consumptions, but is actually fine here! 
b4055aa5aa79bd13394411d42cc37f5c985f34cf